src/prop/prop_engine.cpp

```diff
@@ -593,10 +618,30 @@ std::shared_ptr<ProofNode> PropEngine::getProof()
   {
     return nullptr;
   }
+  if (options::unsatCores()
+      && options::unsatCoresMode() == options::UnsatCoresMode::ASSUMPTIONS)
+  {
+    std::vector<Node> core;
+    getUnsatCore(core);
+    CDProof cdp(d_pnm);
+    Node fnode = NodeManager::currentNM()->mkConst(false);
+    cdp.addStep(fnode, PfRule::SAT_REFUTATION, core, {});
+    return cdp.getProofFor(fnode);
+  }
```

Comment:
I think this should not be in `getProof`, but rather in a different call in the prop engine, like `getRefutation`. As it is if one is in both unsat core and proof mode the actual prop engine proof would never be obtained. So the distinction would be made actually in `SmtEngine::getUnsatCoreInternal`. 

The same applies to the changes in `convertAndAssert` above, which is also making it impossible to get both proofs and unsat cores (from assumptions).

Reply:
You can't call `convertAndAssert` if you do assumption-based unsat cores.

Reply:
Hum, so it should be the case that assumption-based unsat cores are incompatible with proofs. That's fine, we can issue warnings in `set_defaults`.